### ISSUE TITLE
JSDoc @param quick fix to rename reformats entire comment

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#47936]
```js
/**
 * 
 * @param eks {string} the first param
 * @param y {number} the second param
 */
function foo(x, y) {
}
```

Use the `Rename '@param' tag name 'eks' to 'x'` refactoring.

**Expected**

```js
/**
 * 
 * @param x {string} the first param
 * @param y {number} the second param
 */
function foo(x, y) {
}
```

**Actual**

```js
/**
 * @param {string} x the first param
 * @param {number} y the second param
 */
function foo(x, y) {
}
```

Notice that

* A newline was deleted in the comment
* The parameter names and types are swapped

**Comment by a-tarasyuk** [comment]
I think this issue is related to https://github.com/microsoft/TypeScript/issues/47923. That's because we replace the entire JSDoc comment with new changes. We use this approach in all places where we need to add/remove tags from/to JSDoc. [fixOverrideModifier](https://github.com/microsoft/TypeScript/blob/2e706758af0184adc4e6580b60427b1b24a824e2/src/services/codefixes/fixOverrideModifier.ts#L139), [inferFromUsage](https://github.com/microsoft/TypeScript/blob/c44fd466a7c559c363a486169241d89d936ae93d/src/services/codefixes/inferFromUsage.ts#L134), [fixOverrideModifier](https://github.com/microsoft/TypeScript/blob/2e706758af0184adc4e6580b60427b1b24a824e2/src/services/codefixes/fixOverrideModifier.ts#L139), [fixUnmatchedParameter](https://github.com/microsoft/TypeScript/blob/b456702755f86bb2e5c814e2f166f34880fe2102/src/services/codefixes/fixUnmatchedParameter.ts#L37) etc. I think we need to extend [textChanges](https://github.com/microsoft/TypeScript/blob/main/src/services/textChanges.ts) to correctly work with JSDoc nodes. @sandersn What do you think?

**Comment by sandersn** [comment]
Yes, I believe that is the reason -- although #47923 is a lot more severe since it mangles the semantics of jsdoc when it drops the newline.

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
